features/feature_case/asm/asm.cu

```diff
@@ -0,0 +1,1410 @@
+// ====------ asm.cu -------------------------------------- *- CUDA -* ---===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//
+// ===--------------------------------------------------------------------===//
+
```

Comment:
As this file is quite big, ~1.5K line, please summary the test plan in the header part.